lib/jxl/encode.cc

```diff
@@ -40,10 +40,25 @@ uint32_t JxlEncoderVersion(void) {
 JxlEncoderStatus JxlEncoderStruct::RefillOutputByteQueue() {
   jxl::MemoryManagerUniquePtr<jxl::JxlEncoderQueuedFrame> input_frame =
       std::move(input_frame_queue[0]);
+
   input_frame_queue.erase(input_frame_queue.begin());
 
-  // TODO(zond): If the frame queue is empty and the input_closed is true,
-  // then mark this frame as the last.
+  // If the frame queue is empty and input_closed is true,
+  // mark this frame as the last.
+  jxl::FrameInfo frame_info;
+  if (input_closed && input_frame_queue.empty()) {
+    frame_info.is_last = true;
+  } else {
+    frame_info.is_last = false;
+  }
+
+  // don't encode more than one non-animation frame,
+  // until API is able to decode them
```

Comment:
What do you mean here by API unable to decode them? For non-animation frames, the API is able to decode them, they can for example be frames that are blended together to form a single still image. The decoder API only returns the final blended frame as intended by default.

The decode API doesn't currently have an option to return the individual (pre-blended) non-animation frames. Such option could be added (e.g. for more low-level decoding with access to this), but it's not the default option: the intended goal of multiple non-animation frames is to display a single still image result, so it behaves correctly.

Reply:
The API and `djxl` behave differently.  `djxl` saves each frame to a separate file.

Reply:
I changed the comment: until the decode API can return them or the encode API can set processing parameters